types/index.d.ts

```diff
@@ -107,25 +107,38 @@ declare namespace Realm {
         error?: ErrorCallback;
     }
 
-    /**
-     * realm configuration
-     * @see { @link https://realm.io/docs/javascript/latest/api/Realm.html#~Configuration }
-     */
-    interface Configuration {
+    interface BaseConfiguration {
         encryptionKey?: ArrayBuffer | ArrayBufferView | Int8Array;
-        migration?: MigrationCallback;
+        schema?: (ObjectClass | ObjectSchema)[];
+        schemaVersion?: number;
         shouldCompactOnLaunch?: (totalBytes: number, usedBytes: number) => boolean;
         path?: string;
         fifoFilesFallbackPath?: string;
         readOnly?: boolean;
+    }
+
+    interface ConfigurationWithSync extends BaseConfiguration {
+        sync: SyncConfiguration,
+        migration?: never,
+        inMemory?: never,
+        deleteRealmIfMigrationNeeded?: never,
+        disableFormatUpgrade?: never
+    }
+
+    interface ConfigurationWithoutSync extends BaseConfiguration {
+        sync?: never,
```

Comment:
```suggestion
        sync: SyncConfiguration;
        migration?: never;
        inMemory?: never;
        deleteRealmIfMigrationNeeded?: never;
        disableFormatUpgrade?: never;
    }

    interface ConfigurationWithoutSync extends BaseConfiguration {
        sync?: never;
```
The interface properties should end in `;`